x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/ScoreScriptUtilsTests.java

```diff
@@ -39,72 +40,138 @@ public void testDenseVectorFunctions() {
+
+        ScoreScript scoreScript = mock(ScoreScript.class);
+        when(scoreScript._getIndexVersion()).thenReturn(indexVersion);
+
+        List<Number> queryVector = Arrays.asList(0.5f, 111.3f, -13.0f, 14.8f, -156.0f);
+
+        // test dotProduct
+        DotProduct dotProduct = new DotProduct(scoreScript, queryVector, dvs);
+        double result = dotProduct.dotProduct();
+        assertEquals("dotProduct result is not equal to the expected value!", 65425.624, result, 0.001);
+
+        // test cosineSimilarity
+        CosineSimilarity cosineSimilarity = new CosineSimilarity(scoreScript, queryVector, dvs);
+        double result2 = cosineSimilarity.cosineSimilarity();
+        assertEquals("cosineSimilarity result is not equal to the expected value!", 0.790, result2, 0.001);
+
+        // test l1Norm
+        L1Norm l1norm = new L1Norm(scoreScript, queryVector, dvs);
+        double result3 = l1norm.l1norm();
+        assertEquals("l1norm result is not equal to the expected value!", 485.184, result3, 0.001);
+
+        // test l2norm
+        L2Norm l2norm = new L2Norm(scoreScript, queryVector, dvs);
+        double result4 = l2norm.l2norm();
+        assertEquals("l2norm result is not equal to the expected value!", 301.361, result4, 0.001);
+
+        // test dotProduct fails when queryVector has wrong number of dims
+        List<Number> invalidQueryVector = Arrays.asList(0.5, 111.3);
+        DotProduct dotProduct2 = new DotProduct(scoreScript, invalidQueryVector, dvs);
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, dotProduct2::dotProduct);
+        assertThat(e.getMessage(), containsString("query vector has a different number of dimensions [2] than the document vectors [5]"));
+
+        // test cosineSimilarity fails when queryVector has wrong number of dims
+        CosineSimilarity cosineSimilarity2 = new CosineSimilarity(scoreScript, invalidQueryVector, dvs);
+        e = expectThrows(IllegalArgumentException.class, cosineSimilarity2::cosineSimilarity);
+        assertThat(e.getMessage(), containsString("query vector has a different number of dimensions [2] than the document vectors [5]"));
+
+        // test l1norm fails when queryVector has wrong number of dims
+        L1Norm l1norm2 = new L1Norm(scoreScript, invalidQueryVector, dvs);
+        e = expectThrows(IllegalArgumentException.class, l1norm2::l1norm);
+        assertThat(e.getMessage(), containsString("query vector has a different number of dimensions [2] than the document vectors [5]"));
+
+        // test l2norm fails when queryVector has wrong number of dims
+        L2Norm l2norm2 = new L2Norm(scoreScript, invalidQueryVector, dvs);
+        e = expectThrows(IllegalArgumentException.class, l2norm2::l2norm);
+        assertThat(e.getMessage(), containsString("query vector has a different number of dimensions [2] than the document vectors [5]"));
+
+        assertWarnings(ScoreScriptUtils.DEPRECATION_MESSAGE);
```

Comment:
Can we split these tests into separate test functions? Each should be asserting a warning is being emitted, instead of this one check at the end that only guarantees one of them had a deprecation warning?

Reply:
For this PR, I made sure to `assertWarnings` after every function invocation. I plan to refactor this test in a follow-up PR (as it's a reasonably big change).

Reply:
I opened a follow-up PR here: https://github.com/elastic/elasticsearch/pull/48662